File: 1-two-sum/two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var hash = [Int: Int]()
        for (index, n) in nums.enumerated() {
            if hash[target - n] != nil, let lower = hash[target - n] {
                return [lower, index]
            } else {
                hash[n] = index
            }
        }
        return []
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}

        for i in range(len(nums)):
            num = nums[i]
            if num not in hashmap:
                hashmap[target - num] = i
            else:
                return [
                    hashmap[num], i
                ]
        return None